spring/src/main/java/org/axonframework/spring/config/annotation/SpringBeanDependencyResolverFactory.java

```diff
@@ -0,0 +1,102 @@
+/*
+ * Copyright (c) 2010-2020. Axon Framework
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *    http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.axonframework.spring.config.annotation;
+
+import org.axonframework.common.Priority;
+import org.axonframework.eventhandling.AnnotationEventHandlerAdapter;
+import org.axonframework.messaging.Message;
+import org.axonframework.messaging.annotation.ParameterResolver;
+import org.axonframework.messaging.annotation.ParameterResolverFactory;
+import org.slf4j.Logger;
+import org.slf4j.LoggerFactory;
+import org.springframework.beans.BeansException;
+import org.springframework.beans.factory.annotation.Autowired;
+import org.springframework.beans.factory.config.AutowireCapableBeanFactory;
+import org.springframework.beans.factory.config.DependencyDescriptor;
+import org.springframework.context.ApplicationContext;
+import org.springframework.context.ApplicationContextAware;
+import org.springframework.core.MethodParameter;
+import org.springframework.lang.NonNull;
+
+import java.lang.reflect.Constructor;
+import java.lang.reflect.Executable;
+import java.lang.reflect.Method;
+import java.lang.reflect.Parameter;
+import java.util.concurrent.atomic.AtomicReference;
+
+import static org.axonframework.eventhandling.GenericEventMessage.asEventMessage;
+
+/**
+ * ParameterResolverFactory implementation that resolves parameters using Spring dependency resolution.
```

Comment:
Just to get my head straight, but this approach also allows the `@Qualifier` annotation on the parameters, since Spring's `DependencyDescriptor` simply includes that logic, correct?
If so, it would be nice to see a test case for that combination in the `SpringBeanParameterResolverFactoryTest` class.
Next to calling this out in the JavaDoc :-)

Reply:
This approach allows _everything_ that Spring is capable of on the parameters.  I don't think it makes sense to check every Spring annotation since you'd be retesting the wheel.  The `NoUniqueBeanDefinitionException` test ensures that it is in fact using Spring dependency resolution.

Reply:
Awesome, then my mind was serving me right.
Fair point on testing, going the stretch to do _everything_ is definitely overboard.
I was pointing out the `@Qualifier` here as we supported this already and I would like to make sure that support is maintained/taken over by your implementation.